### ISSUE TITLE
Design Forward: Search - redesign to filter in app + results from other apps

### ISSUE THREAD

**Issue by jancborchardt** [owncloud/core#821]
Currently the search is only global, and has a small popup detached from the rest of the interface. It’s not really possible to filter contents from the local app, which will get really impractical with lots of files or other very text-based apps like Mail, Notes etc.

@butonic and I discussed a better approach for this:
The search should prioritize the currently used app. That is done by live-filtering the content (like Gmail, Dropbox, etc do it). Then below these results of the current app, related results from other apps can be shown. So when you search for a document and come to the end of the list (which might be only 3 entries) there will be similarly named calendar entries, contacts, and so on – visually distinguished of course.

Mockups to follow. What do you think about the general idea, especially in comparison to the current approach?

**Comment by MorrisJobke** [comment]
@jancborchardt This is fixed once #12653 is merged, or am I wrong?

**Comment by butonic** [comment]
Partly. We should still introduce a query parser. But that can be tracked in a new issue.

**Comment by MorrisJobke** [comment]
@butonic Can you take care of this? Thanks

**Comment by butonic** [comment]
query parser tracked in https://github.com/owncloud/core/issues/13155
